src/stratigraphy/benchmark/score.py

```diff
@@ -379,23 +182,102 @@ def create_predictions_objects(
+
+    if input_directory and draw_directory:
+        draw_predictions(predictions, input_directory, draw_directory)
+
+    if number_of_truth_values:  # only evaluate if ground truth is available
+        metrics = evaluate_borehole_extraction(predictions, number_of_truth_values)
+
+        all_series = [
+            metrics["layer"].to_dataframe("F1", lambda metric: metric.f1),
+            metrics["layer"].to_dataframe("precision", lambda metric: metric.precision),
+            metrics["layer"].to_dataframe("recall", lambda metric: metric.recall),
+            metrics["depth_interval"].to_dataframe("Depth_interval_accuracy", lambda metric: metric.precision),
+            metrics["layer"].to_dataframe("Number Elements", lambda metric: metric.tp + metric.fn),
+            metrics["layer"].to_dataframe("Number wrong elements", lambda metric: metric.fp + metric.fn),
+            metrics["coordinates"].to_dataframe("coordinates", lambda metric: metric.f1),
+            metrics["elevation"].to_dataframe("elevation", lambda metric: metric.f1),
+            metrics["groundwater"].to_dataframe("groundwater", lambda metric: metric.f1),
+            metrics["groundwater_depth"].to_dataframe("groundwater_depth", lambda metric: metric.f1),
+        ]
+        document_level_metrics = pd.DataFrame()
+        for series in all_series:
+            document_level_metrics = document_level_metrics.join(series, how="outer")
+
+        document_level_metrics.to_csv(
+            temp_directory / "document_level_metrics.csv", index_label="document_name"
+        )  # mlflow.log_artifact expects a file
+
+        coordinates_metrics = metrics["coordinates"].overall_metrics()
+        groundwater_metrics = metrics["groundwater"].overall_metrics()
+        groundwater_depth_metrics = metrics["groundwater_depth"].overall_metrics()
+        elevation_metrics = metrics["elevation"].overall_metrics()
+
+        metrics_dict = {
+            "F1": metrics["layer"].pseudo_macro_f1(),
+            "recall": metrics["layer"].macro_recall(),
+            "precision": metrics["layer"].macro_precision(),
+            "depth_interval_accuracy": metrics["depth_interval"].macro_precision(),
+            "de_F1": metrics["de_layer"].pseudo_macro_f1(),
+            "de_recall": metrics["de_layer"].macro_recall(),
+            "de_precision": metrics["de_layer"].macro_precision(),
+            "de_depth_interval_accuracy": metrics["de_depth_interval"].macro_precision(),
+            "fr_F1": metrics["fr_layer"].pseudo_macro_f1(),
+            "fr_recall": metrics["fr_layer"].macro_recall(),
+            "fr_precision": metrics["fr_layer"].macro_precision(),
+            "fr_depth_interval_accuracy": metrics["fr_depth_interval"].macro_precision(),
+            "coordinate_f1": coordinates_metrics.f1,
+            "coordinate_recall": coordinates_metrics.recall,
+            "coordinate_precision": coordinates_metrics.precision,
+            "groundwater_f1": groundwater_metrics.f1,
```

Comment:
I introduced the notion of groundwater and groundwater depth. However, I do think that this duplicates a lot of the code, and we are mostly interested in the depth of the groundwater, correct? Therefore, I would suggest to delete the metrics for the entire groundwater object.

Reply:
I tend to disagree, I would say that the elevation and especially the measurement date are also relevant. So either we add metrics for those separately, of we compute a metric for the whole groundwater object, as we currently do.

Reply:
I think you are right when it comes to the date. I had only the elevation of the groundwater in mind. Then we can keep things like this.